### PR TITLE
Folia support

### PR DESCRIPTION
**Pull request by kugge**
Add [Folia](https://github.com/PaperMC/Folia) support.

* Lag checker and watchdog don't work on Folia so they're disabled by default
* Use reflection to avoir changing API (if you want me to convert it to Folia API ping me)
* Need more tests

### CODE REVIEW

**Review by Vankka — CHANGES_REQUESTED** [review]
I left a few in-line comments/style fixes

**Comment by kugge** [comment]
Thank you for your review. I force pushed the requested changes. I believe that more tests are needed, though I haven't encountered any problem yet.

**Comment by Vankka** [comment]
Kicking players does appear to work on the global scheduler but it should probably occur on the Player's own EntityScheduler (Entity#getScheduler). The other scheduler calls look to be using the appropriate scheduler(s)

Otherwise I'm fine with merging this

**Comment by kugge** [comment]
> Kicking players does appear to work on the global scheduler but it should probably occur on the Player's own EntityScheduler (Entity#getScheduler). The other scheduler calls look to be using the appropriate scheduler(s)
> 
> Otherwise I'm fine with merging this

Done, changes were force-pushed

File: src/main/java/github/scarsz/discordsrv/listeners/DiscordBanListener.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ public void onGuildBan(GuildBanEvent event) {
         if (offlinePlayer.isOnline()) {
             // also kick them because adding them to the BanList isn't enough
             Player player = offlinePlayer.getPlayer();
-            if (player != null) Bukkit.getScheduler().runTask(DiscordSRV.getPlugin(), () -> player.kickPlayer(reason));
+            if (player != null) SchedulerUtil.runTask(DiscordSRV.getPlugin(), () -> player.kickPlayer(reason));
```
Comment (Vankka) [inline src/main/java/github/scarsz/discordsrv/listeners/DiscordBanListener.java:62]: This needs to use the new method as well

Comment (kugge) [reply src/main/java/github/scarsz/discordsrv/listeners/DiscordBanListener.java]: Done. Don't hesitate to ping me if I've forgotten something or if tests fails.

**Comment by kugge** [comment]
Added Scheduler caching
Will consider adding Method Handlers

**Review by Vankka — APPROVED** [review]

**Pull request merged**